src/frontend/js/components/PaymentInterfaces/index.spec.tsx

```diff
@@ -0,0 +1,67 @@
+import { ContextFactory as mockContextFactory, PaymentFactory } from 'utils/test/factories';
+import { render, screen } from '@testing-library/react';
+import { handle as mockHandle } from 'utils/errors/handle';
+import * as Joanie from 'types/Joanie';
+import { PaymentProviders } from 'types/Joanie';
+import { noop } from 'utils';
+import PaymentProvider from '.';
+
+jest.mock('./PayplugLightbox', () => ({
+  __esModule: true,
+  default: () => 'Payplug lightbox',
+}));
+jest.mock('./Dummy', () => ({
+  __esModule: true,
+  default: () => 'Dummy payment component',
+}));
+
+jest.mock('utils/errors/handle', () => ({
+  handle: jest.fn(),
+}));
+
+jest.mock('utils/context', () => ({
+  __esModule: true,
+  default: mockContextFactory().generate(),
+}));
+
+describe('PaymentProvider', () => {
+  it('should return null and handle an error if payment provider is not implemented', () => {
+    const payment = PaymentFactory.afterGenerate((p: Joanie.Payment) => ({
+      ...p,
+      provider: 'unknown-provider',
+    })).generate();
+
+    const onSuccess = jest.fn();
+    const onError = jest.fn();
+
+    const { container } = render(
+      <PaymentProvider {...payment} onSuccess={onSuccess} onError={onError} />,
+    );
+
+    expect(mockHandle).toHaveBeenCalledWith(
+      new Error('Payment provider unknown-provider not implemented'),
+    );
+    expect(onError).toHaveBeenCalledWith('errorDefault');
+    expect(container.childElementCount).toEqual(0);
+  });
+
+  it('should render the payplug lightbox when provider is Payplug', async () => {
+    const payment = PaymentFactory.afterGenerate((p: Joanie.Payment) => ({
+      ...p,
+      provider: PaymentProviders.PAYPLUG,
+    })).generate();
+    render(<PaymentProvider onSuccess={noop} onError={noop} {...payment} />);
+
```

Comment:
just to be certain I would make sure on both tests that onError hasn't been called